docs/concepts/promptflow-vs-langchain.md

```diff
@@ -0,0 +1,49 @@
+* Iterate on your application flow and run evaluation flows via the SDK/CLI, allowing you to compare metrics and choose the optimal candidate for release. These iterations include trying different prompts, different LLM parameters like temperature etc. - this is referred as “tuning” process earlier, or sometime referred as experimentation.
+* Integrate the evaluation into your CI/CD pipeline, aligning the assertions in your gated tests with the selected metrics.
+
+
+Prompt Flow introduces two conceptual components to facilitate this workflow:
+
+* Evaluation flow: a flow type that indicates this flow is not for deploy or integrate into your app, it’s for evaluating an app/flow performance.
+* Run: every time you run your flow with data, or run an evaluation on the output of a flow, a Run object is created to manage the history and allow for comparison and additional analysis.
+
+While new concepts introduce additional cognitive load, we firmly believe they hold greater importance compared to abstracting different LLM APIs or vector database APIs.
+
+## 3. Optimize for “visibility”.
+
+There are quite some interesting application patterns emerging because of LLMs, like Retrieval Augmented Generation (RAG), ReAct and more. While how LLMs work may remain enigmatic to many developers, how LLM apps work is not - they essentially involve a series of calls to external services such as LLMs, databases, and search engines, all glued together. Architecturally there isn’t much new, patterns like RAG and ReAct are both straightforward to implement once a developer understands what they are - plain Python programs with API calls to external services can totally serve the purpose effectively.
+
+By observing many internal use cases, we learned that deeper insight into the detail of the execution is critical. Establishing a systematic method for tracking interactions with external systems is one of design priority. Consequently, We adopted an unconventional approach - prompt flow has a YAML file describing how function calls (we call them [Tools](../concepts/concept-tools.md)) are executed and connected into a Directed Acyclic Graph (DAG). 
+
+This approach offers several key benefits, primarily centered around enhanced **visibility**:
+1) During development, your flow can be visualized in an intelligible manner, enabling clear identification of any faulty components. As a byproduct, you obtain an architecturally descriptive diagram that can be shared with others.
+2) Each node in the flow has it’s internal detail visualized in a consistent way.
+3) Single nodes can be individually run or debugged without the need to rerun previous nodes.
+
+The emphasis on visibility in Prompt flow's design helps developers to gain a comprehensive understanding of the intricate details of their applications, ultimately providing effective troubleshooting and optimization.
+</b>
+
+![promptflow-dag](../media/promptflow-dag.png)
```

Comment:
For completeness it would be good to also talk about the tradeoff of this approach - like even we add some control flow constructs like "activate-when" to serve the needs of branches / switch-case, we're not going to make the Flow itself Turning complete. If you want to develop an agent which is totally dynamic and driven by LLM, it might be good to use Semantic Kernel together with prompt flow.